pkg/collector/corechecks/cluster/ksm/customresources/job.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package customresources
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kube-state-metrics/v2/pkg/customresource"
+	"k8s.io/kube-state-metrics/v2/pkg/metric"
+	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
+)
+
+var descJobLabelsDefaultLabels = []string{"namespace", "job_name"}
+
+// NewJobFactory returns a new Job metric family generator factory.
+func NewJobFactory() customresource.RegistryFactory {
+	return &jobFactory{}
+}
+
+type jobFactory struct{}
+
+func (f *jobFactory) Name() string {
+	return "jobs_extended"
+}
+
+// CreateClient is not implemented
+func (f *jobFactory) CreateClient(cfg *rest.Config) (interface{}, error) {
+	panic("not implemented")
+}
+
+func (f *jobFactory) MetricFamilyGenerators(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
+	return []generator.FamilyGenerator{
+		*generator.NewFamilyGenerator(
+			"kube_job_duration",
+			"Duration represents the time elapsed between the StartTime and CompletionTime of a Job",
```

Comment:
Looks like the description doesn't reflect the behavior of the metric when the job is in running (i.e when `CompletionTime == nil`)
So either we make the description more explicit or we send the metric only when the job is done (i.e when `CompletionTime != nil`). No strong opinion but I find the latter more intuitive.